.github/workflows/tests.yml

```diff
@@ -1,27 +1,120 @@
-name: Tests
+name: CI Workflow for tests
 
 on:
+  pull_request_target:
+    types: [opened, synchronize, reopened, labeled]
   push:
-    branches:
-      - master
-  pull_request:
     branches: [ master ]
 
-concurrency:
-  group: ci-${{ github.ref }}-test
-  cancel-in-progress: true
-
 jobs:
+  permission:
+    if: ${{ github.event_name == 'pull_request_target' }}
+    strategy:
+      matrix:
+        platform: [macos-latest, windows-latest, ubuntu-latest]
```

Comment:
I think a matrix is overkill here, and a simple `ubuntu-latest` should be enough, no?